script.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+
+log_file=""
+method=1
+country_flag=false
+
+# Help 
+show_help() {
+  echo "Usage: $0 [-f <file>] [-m <method>] [-c] [-h]"
+  echo ""
+  echo "Options:"
+  echo "  -f, --file     Path to the Nginx log file"
+  echo "  -m, --method   Parsing method (1: awk, 2: grep)"
+  echo "  -c, --country  Fetch country information for each IP"
+  echo "  -h, --help     Show help information"
+  echo ""
+  echo "Examples:"
+  echo "  $0 -f nginx.log -m 1"
+  echo "  $0 -f nginx.log -m 2 -c"
+}
+
+# Перший метод через awk
+parse_with_awk() {
+  echo "Using awk to parse the log file..."
+  awk '{print $1}' "$log_file" | sort | uniq -c | sort -nr | awk '{print $2, $1}'
+}
+
+# Другий метод через grep
+parse_with_grep_cut() {
+  echo "Using grep to parse the log file..."
+  grep -oP '^\d+\.\d+\.\d+\.\d+' "$log_file" | sort | uniq -c | sort -nr | awk '{print $2, $1}'
+}
```

Comment:
I'm not sure about this method. Look what I got when I tried to run the command from your README

```
./script.sh -f nginx.log -m 2 -c
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
Using grep null
```